crates/bevy_ecs/examples/derive_label.rs

```diff
@@ -0,0 +1,39 @@
+use bevy_ecs::prelude::*;
+
+fn main() {
+    // Unit labels are always equal.
+    assert_eq!(UnitLabel.as_label(), UnitLabel.as_label());
+
+    // Enum labels depend on the variant.
+    assert_eq!(EnumLabel::One.as_label(), EnumLabel::One.as_label());
+    assert_ne!(EnumLabel::One.as_label(), EnumLabel::Two.as_label());
+
+    // Shockingly, labels annotated with `ignore_fields` ignore their fields.
+    assert_eq!(WeirdLabel(1).as_label(), WeirdLabel(2).as_label());
+
+    // Labels of different types are distinct, if they look similar.
+    assert_ne!(
+        GenericLabel::<f64>::One.as_label(),
+        GenericLabel::<char>::One.as_label()
+    );
+}
+
+#[derive(SystemLabel)]
+pub struct UnitLabel;
+
+#[derive(SystemLabel)]
+pub enum EnumLabel {
+    One,
+    Two,
+}
+
+#[derive(SystemLabel)]
+#[system_label(ignore_fields)]
+pub struct WeirdLabel(i32);
+
+#[derive(SystemLabel)]
+pub enum GenericLabel<T> {
+    One,
+    #[system_label(ignore_fields)]
```

Comment:
```suggestion
    #[system_label(ignore)]
```

Bikeshedding: I think this is marginally clearer, less verbose and more idiomatic.

Reply:
To me, that implies that we're ignoring the entire variant, but we're only ignoring the fields.

Reply:
Oh I see 🤔 Yeah I think you're right, but we should make sure to make this clear in the docs.

Weird to think that you could set your enum variant to the phantomdata variant...